### PR TITLE
Service portal : Manual filter of external Url documents + Bug fixes

### PR DESCRIPTION
**Pull request by obmagnusson**
# \<Description\>

Attach a link to issue if relevant

## What

Added manual filter of external url documents, 
all other documents are treated as PDF document. 

### Bug fixes
Popup blocker would prevent pdf documents to open in new tab. 
The fix has been tested in these browsers, and supports very slow response time (at least 15 sec) : 

- Safari, mobile + desktop ✅ 
- Firefox ✅ 
- Edge ✅ 
- Opera, 💹 

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [] have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by Thorrik58 — APPROVED** [review]

File: libs/api/domains/documents/src/lib/documentUrlBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@nestjs/common'
+import { Document } from './models/document.model'
+import { DocumentInfoDTO } from './client/models'
+import { DocumentTypeFilter } from './types/documentTypeFilterType'
+import { FileType } from './types/fileType.enum'
+
+@Injectable()
+export class DocumentBuilder {
+  constructor() { }
+
```
Comment (Thorrik58) [inline libs/api/domains/documents/src/lib/documentUrlBuilder.ts:10]:
```suggestion
// Handling edge case for documents that cant be presented due to requiring authentication through rsk.is

```

File: libs/service-portal/documents/src/components/DocumentCard/DocumentCard.tsx [hunk 2]
```diff
@@ -33,46 +43,65 @@ interface Props {
...
+    // Note: opening window before fetching data, to prevent popup-blocker
+    let windowRef = window.open()
+    try {
+      const { data } = await client.query({
+        query: GET_DOCUMENT,
+        variables: { input: { id: document.id } },
+      })
+      const doc = data?.getDocument
+      if (!doc) {
+        throw new Error()
```
Comment (Thorrik58) [inline libs/service-portal/documents/src/components/DocumentCard/DocumentCard.tsx:82]: Should probably have some kind of description of which error is being thrown here.

File: libs/service-portal/documents/src/components/DocumentCard/DocumentCard.tsx [hunk 3]
```diff
@@ -86,7 +115,8 @@ const DocumentCard: FC<Props> = ({ document }) => {
         key={document.id}
         loading={loading}
         cta={{
-          onClick: handleOnClick,
+          externalUrl: getEdgecaseDocument(document)?.url,
```
Comment (Thorrik58) [inline libs/service-portal/documents/src/components/DocumentCard/DocumentCard.tsx:118]: Do we need a fallback in case this is empty? @andriandresson

Comment (andriandresson) [reply libs/service-portal/documents/src/components/DocumentCard/DocumentCard.tsx:118]: `onClick: handleOnClick` is never empty. ActionCard Component provides the logic to render either Link, if externalUrl prop is provided, or Button, and onClick prop is always provided (not optional prop).

**Review by Thorrik58 — APPROVED** [review]

**Pull request merged**